service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -117,15 +131,15 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 		},
 	}
 
-	if clusterProfile >= xs {
+	if determinedTCProfile >= xs {
 		for _, configMapSpec := range configMapSpecs {
 			_, ok := configMapSpec.Values["cluster"]
 			if !ok {
 				configMapSpec.Values["cluster"] = make(map[string]interface{})
 			}
 
 			clusterMap := configMapSpec.Values["cluster"].(map[string]interface{})
-			clusterMap["profile"] = clusterProfile
+			clusterMap["profile"] = determinedTCProfile
```

Comment:
if `determinedTCProfile` is `xxs`, we don't need to set a value from here?

Reply:
Fixed, PTAL

Reply:
Thanks for the fix, so it's same as checking `determinedTCProfile` is not `0`, right?

Reply:
Yes. Thought one can't compare custom named type with ints like 0 directly, but it turns out all int operations are possible for the custom type too. So, I've updated the condition not to use any specific cluster profile as they can change, come and go.